model/cluster_request.go

```diff
@@ -84,6 +84,9 @@ func (request *CreateClusterRequest) SetDefaults() {
 	if _, ok := request.DesiredUtilityVersions[PgbouncerCanonicalName]; !ok {
 		request.DesiredUtilityVersions[PgbouncerCanonicalName] = PgbouncerDefaultVersion
 	}
+	if _, ok := request.DesiredUtilityVersions[StackroxCanonicalName]; !ok {
```

Comment:
Just realized we already had a mechanism in place that we rediscovered again in our discussion 😂 

I just wonder if we should:
- Add a check if `ValuesPath` is not empty if the utility is present. For example:
```go
	if _, ok := request.DesiredUtilityVersions[PgbouncerCanonicalName]; !ok {
		request.DesiredUtilityVersions[PgbouncerCanonicalName] = PgbouncerDefaultVersion
	} else if request.DesiredUtilityVersions[PgbouncerCanonicalName].Values() == "" {
		request.DesiredUtilityVersions[PgbouncerCanonicalName].ValuesPath = PgbouncerDefaultVersion.ValuesPath
	}
```
As if I am not mistaken we are able to pass nonempty `Version` but empty `ValuesPath` which probably would not work.
- I think we should also add set default in `ProvisionClusterRequest` as such:
```go
	pgBouncerUtility, ok := request.DesiredUtilityVersions[PgbouncerCanonicalName]
	if ok && pgBouncerUtility.Values() == ""  {
		request.DesiredUtilityVersions[PgbouncerCanonicalName].ValuesPath = PgbouncerDefaultVersion.ValuesPath
	}
```
So that we do not need to pass the full values path together with the version if we want to upgrade the utility.


I might be wrong on that as I did not test it and going with my gut here 😄

Reply:
We have discussed to resolve it as a follow up https://mattermost.atlassian.net/browse/MM-37273